### PR TITLE
Syntax extensions on trait and impl items.

### PR DESCRIPTION
**Pull request by nrc**
Allows modifiers to be used on methods, associated types, etc.

r? @sfackler

### CODE REVIEW

File: src/libsyntax/ext/base.rs [hunk 1]
```diff
@@ -73,6 +73,108 @@ impl<F> ItemModifier for F
...
 
+#[derive(Show,Clone)]
+pub enum Annotatable {
+    Item(P<ast::Item>),
+    TraitItem(ast::TraitItem),
+    ImplItem(ast::ImplItem),
+}
+
+impl Annotatable {
+    pub fn attrs(&self) -> Vec<ast::Attribute> {
```
Comment (sfackler) [inline src/libsyntax/ext/base.rs]: Why not `&[Attribute]`?

Comment (nick29581) [inline src/libsyntax/ext/base.rs]: It works! (Pretty impressed rustc is able to figure out the lifetimes with that)

**Comment by nick29581** [comment]
Fixed according to comment and rebased.

r? @sfackler

**Comment by sfackler** [comment]
This seems okay for now, until we work out how we actually want to stabilize this stuff.

**Pull request merged**